samples/performance/pipeline_cache/pipeline_cache.cpp

```diff
@@ -43,11 +43,11 @@ PipelineCache::~PipelineCache()
 	{
 		/* Get size of pipeline cache */
 		size_t size{};
-		VK_CHECK(vkGetPipelineCacheData(get_device().get_handle(), pipeline_cache, &size, nullptr));
+		vkGetPipelineCacheData(get_device().get_handle(), pipeline_cache, &size, nullptr);
```

Comment:
Why do you remove the VK_CHECK here?

Reply:
I removed this since I understand that it's never a good idea to `throw()` from a destructor.  I believe that modern C++ defaults for destructors is a non-throwing exception specification.  In addition my IDE (Xcode) warns me about this with its static code parser.  If you are concerned about this sample's case (the only one I can find in the project that calls VK_CHECK in a destructor), I could instead grab the return code and if not success post a fatal error vs. a throw from the VK_CHECK macro.

I am happy to implement you preference here: a) leave it with the VK_CHECK removed, or b) add in a test for failure using the old VK_CHECK expansion code:

```
if (result != VK_SUCCESS)
{
	LOGE("Detected Vulkan error: {}", vkb::to_string(result));
	abort();
}      
```

Reply:
Good catch!

I think, some error check for those cases would be good. Or maybe, you could even introduce something like a `VK_CHECK_NOTHROW`, or so?

Reply:
Exactly. Just ignoring the error isn't good practice either. So your proposal of log + abort looks good to me. Or, since this isn't really related to the PR, just revert this and we'll put it in our pile for the larger framework refactor.

Reply:
Well, it's kind of related since by calling `throw()` inside `VK_CHECK` makes it inappropriate for a destructor.  I don't think I should just revert to original since it would leave a new problem behind.  I had a closer look at what the code is actually doing and would like to suggest the following:

```
/* Get size of pipeline cache */
size_t size{};
VkResult result = vkGetPipelineCacheData(get_device().get_handle(), pipeline_cache, &size, nullptr);
if (result == VK_SUCCESS)
{
	/* Get data of pipeline cache */
	std::vector<uint8_t> data(size);
	result = vkGetPipelineCacheData(get_device().get_handle(), pipeline_cache, &size, data.data());
	if (result == VK_SUCCESS)
	{
		/* Write pipeline cache data to a file in binary format */
		vkb::fs::write_temp(data, "pipeline_cache.data");
	}
}

if (result != VK_SUCCESS)
{
	LOGE("Detected Vulkan error: {}", vkb::to_string(result));
}

/* Destroy Vulkan pipeline cache */
vkDestroyPipelineCache(get_device().get_handle(), pipeline_cache, nullptr);
```

This would detect any errors and prevent the write operation if `vkGetPipelineCacheData()` failed.  It also would report any such failure to the log without doing an abort and then shutdown properly.

Reply:
Submitted _pipeline_cache_ destructor error handling change.  Comments welcome on review.